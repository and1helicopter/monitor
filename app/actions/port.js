// @flow
import type { GetState, Dispatch } from '../reducers/types';

export const INITIALIZE_PORT = 'INITIALIZE_PORT';

export function initialize_port() {
  return {
    type: INITIALIZE_PORT
  };
}

// export function portAsync(delay: number = 1000) {
//   return (dispatch: Dispatch) => {
//     setTimeout(() => {
//       dispatch(port());
//     }, delay);
//   };
// }
