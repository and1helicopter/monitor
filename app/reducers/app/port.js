// import { INITIALIZE_PORT } from '../actions/port';
// import type { Action } from './types';

const initialState = {
    port: "COM1",
    addr: 1,
    baudRate: 115200,
    dataBits: 8, /* 5, 6, 7 */
    stopBits: 1, /* 1.5, 2 */
    parity: 'odd'    
}

export default function port(state: object = initialState, action: Action) {
    switch (action.type){
        // case INITIALIZE_PORT:
        //     return state = object;
        default: 
            return state;
    }
 }