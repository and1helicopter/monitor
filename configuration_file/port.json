{
    "port": "COM4",
    "addr": 0,
    "baudRate": 115200,
    "dataBits": 8,
    "stopBits": 1,
    "parity": "odd"    
}