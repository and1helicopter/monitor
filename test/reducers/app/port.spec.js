import port  from '../../../app/reducers/app/port';

describe('reducers', () => {
    describe('port', () => {
        it('should handle initial state', () => {
            expect(port(undefined, {})).toEqual({    
                port: "COM1",
                addr: 1,
                baudRate: 115200,
                dataBits: 8, 
                stopBits: 1,  
                parity: 'odd'    
            });
        });

        it('should handle somthing', () => {
            expect(port({"port":"COM2"}, {})).toEqual({"port":"COM2"});
        });
    });
});