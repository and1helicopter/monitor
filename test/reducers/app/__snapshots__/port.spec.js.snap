// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers port should handle initial state 1`] = `
Object {
  "addr": 1,
  "baudRate": 115200,
  "dataBits": 8,
  "parity": "odd",
  "port": "COM1",
  "stopBits": 1,
}
`;
